tensorflow/python/framework/python_op_gen.cc

```diff
@@ -367,6 +405,155 @@ string GenEagerPythonOp::Code() {
   return prelude_ + result_;
 }
 
+string GenEagerPythonOp::GetTypeAnnotatedParams() {
+  // holds mappings from param name to its type annotation
+  std::unordered_map<string, string> param_type_map;
+  for (int i = 0; i<op_def_.attr_size(); ++i) {
+    const auto& attr(op_def_.attr(i));
+    if (attr.type() == "type") {
+      bool has_dtype_half = false;
+      for (int t : attr.allowed_values().list().type()) {
+        if (t == 19) { // DT_HALF = 19
+          has_dtype_half = true;
+          break;
+        }
+      }
+
+      // Do not add type annotations to an arg that can be of type half
```

Comment:
Let's write this as a // TODO: remove this once half is supported.

Reply:
Half will be supported with the changes in https://github.com/tensorflow/tensorflow/pull/40696

Reply:
I know that, but the idea is to make a note of that in the code. Otherwise future readers will think there is a special reason not to use these.